packages/react-day-picker/src/contexts/SelectRange/SelectRangeContext.tsx

```diff
@@ -118,63 +118,60 @@ export function SelectRangeProviderInternal({
   if (selected) {
     if (selected.from) {
       modifiers.range_start = [selected.from];
-      if (selected.to) {
-        modifiers.range_middle = [
-          {
-            after: selected.from,
-            before: selected.to
-          }
-        ];
-        if (max || min) {
-          modifiers.disabled = [
-            (date: Date) => {
-              if (
-                max &&
-                selected.to &&
-                selected.from &&
-                isBefore(date, selected.from)
-              ) {
-                const diff = differenceInCalendarDays(selected.to, date);
-                if (diff >= max) {
-                  return true;
-                }
+      selected.to = !selected.to ? selected.from : selected.to;
```

Comment:
I'm setting `selected.to` as the `from` date if it's undefined, while `to` is defined. This is as a result of changing the output of the range context to include `{ from: date, to: undefined }` on the first click.

Reply:
That seems kind of weird to me. You are mutating shared state?

Reply:
That seems like an unwanted side-effect

Reply:
Thank you yes, I'll use a variable instead.